pulumi/infra/kafka.py

```diff
@@ -1,13 +1,137 @@
-from typing import Optional
+from __future__ import annotations
 
-import pulumi_kafka as kafka
+import dataclasses
+import os
+from typing import Any, Iterable, Mapping, Optional, Sequence, Tuple, cast
+
+from infra.config import LOCAL_GRAPL
+from pulumi.stack_reference import StackReference
 
 import pulumi
 
+# this list of service names must match those in the
+# confluent-cloud-infrastructure project:
+# TODO: link
+KAFKA_SERVICES = [
+    "pipeline-ingress",
+    "graph-generator",
+    "node-identifier",
+    "graph-merger",
+    "analyzer-executor",
+    "engagement-creator",
+]
+
+
+@dataclasses.dataclass
+class CredentialOutput:
+    service_account_id: str
+    api_key: str
+    api_secret: str
+
+    @staticmethod
+    def from_json(json_: Mapping[str, str]) -> CredentialOutput:
```

Comment:
Could we call it `input` instead of `json_`? The trailing underscore looks weird here 😅 

(Likewise elsewhere in this PR.)

Reply:
`input` is also a thing in python so it would need a trailing underscore as well (I mean not strictly... you can re-bind any name you want, within the bounds of syntax, but it's not usually advisable).. I'll think of something.

Reply:
`json_input`, to cover all our bases 😅 

It's not a big deal either way, though; whatever you end up deciding is cool 👍